### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by Gudahtt**
A PR template has been added to describe what we want from PRs. It asks for a clear description of changes, references to related issues/PRs, and examples of the change being suggested.

### CODE REVIEW

**Comment by Gudahtt** [comment]
This depends upon #162

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+* See: #67890
+-->
+
+## Examples
+
+<!--
+Are there any examples of this change being used in another repository?
+
+When considering changes to the MetaMask module template, it's strongly preferred that the change be experimented with in another repository first. This gives reviewers a better sense of how the change works, making it less likely the change will need to be reverted or adjusted later.
+-->
```
Comment (Mrtenz) [inline .github/pull_request_template.md:19]: Can we add a section for breaking changes too?

Comment (Gudahtt) [reply .github/pull_request_template.md:19]: Hmm. I considered this, but I wasn't sure how to characterize a "breaking change" for the template. Wouldn't any change be "breaking" in a sense?

**Review by mcmire — DISMISSED** [review]
Looks good!

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<!--
+Thanks for your contribution! Take a moment to answer these questions so that reviewers have the information they need to properly understand your changes:
+
+* What is the current state of things and why does it need to change?
+* What is the solution your changes offer and how does it work?
+
+Are there any issues, Slack conversations, Zendesk issues, user stories, etc. reviewers should consult to understand this pull request better? For instance:
```
Comment (legobeat) [inline .github/pull_request_template.md:7]: My perspective: Public PRs should not include internal links that need privileged access - some of these often are so I don't feel good about effectively normalizing that. This template will also be used by external contributors, often as a first point of interaction.
```suggestion
Are there any issues or other links reviewers should consult to understand this pull request better? For instance:
```

Comment (Gudahtt) [reply .github/pull_request_template.md:7]: Providing sufficient explanation in the PR to understand a change _without_ any additional context should be a goal ideally. But I don't see the harm in adding additional context, even if access is restricted. Pretending the relevant resources don't exist will not help anyone.

Regardless, your suggested change here looks good to me. This was just copied from one of our other templates.

Comment (legobeat) [reply .github/pull_request_template.md]: I think if it's public information, it should be made accessible for readers of the PR. Useful context from Slack can be copy-pasted preferred to make it searchable) or shared as a screenshot, for example.

If it's confidential, it should not be linked.

**Review by mcmire — APPROVED** [review]

**Review by legobeat — APPROVED** [review]

**Pull request merged**